server/app/services/program/ProgramServiceImpl.java

```diff
@@ -862,6 +872,10 @@ public ProgramDefinition removeBlockEligibilityPredicate(long programId, long bl
     } catch (IllegalPredicateOrderingException e) {
       // Removing a predicate should never invalidate another.
       throw new RuntimeException("Unexpected error: removing this predicate invalidates another");
+    } catch (EligibilityNotValidForProgramTypeException e) {
+      // Removing eligibility predicates should always be valid.
+      throw new RuntimeException(
+          "Unexpected error: removing this predicate is not allowed for this ProgramType");
```

Comment:
**nitpick (blocking):** include the original exception so we can see it in the stack trace
```suggestion
      throw new RuntimeException(
          "Unexpected error: removing this predicate is not allowed for this ProgramType", e);
```